src/extract.ts

```diff
@@ -464,6 +497,55 @@ function extractCatch2Result(output: string): BenchmarkResult[] {
     return ret;
 }
 
+function extractJuliaBenchmarkHelper(g: JuliaBenchmarkGroup, labels: string[] = []): BenchmarkResult[] {
+    const res: BenchmarkResult[] = [];
+    for (const key in g[1].data) {
+        const value = g[1].data[key];
+        if (value[0] === 'BenchmarkGroup') {
+            for (const x of extractJuliaBenchmarkHelper(value as JuliaBenchmarkGroup, [...labels, key])) {
+                res.push(x);
+            }
```

Comment:
with the discriminated union type TS should be smart enough so you don't need to cast. Additionally `push` can handle multiple items at once so the loop can be simplified

```suggestion
            res.push(...extractJuliaBenchmarkHelper(value, [...labels, key]))
```